app/assets/javascripts/dashboard.js

```diff
@@ -38,13 +38,22 @@ MinionPoller = {
         var minions = data.assigned_minions || [];
         var unassignedMinions = data.unassigned_minions || [];
 
-        MinionPoller.selectedMasters = data.assigned_minions.reduce(function(memo, minion) {
-          if (minion.role === 'master') {
-            memo.push(minion.id);
-          }
-
-          return memo;
-        }, []);
+        // for the dashboard, if we rely on radio, the first time this comes
+        // it won't detect that there's a master, so we need to rely on the data
+        // itself for counting masters
+        if (MinionPoller.renderMode === 'dashboard') {
+          MinionPoller.selectedMasters = data.assigned_minions.reduce(function(memo, minion) {
+            if (minion.role === 'master') {
+              memo.push(minion.id);
+            }
+
+            return memo;
+          }, []);
+        } else {
+          MinionPoller.selectedMasters = $("input[name='roles[master][]']:checked").map(function() {
+            return parseInt($( this ).val());
```

Comment:
No. It was like that before. I can change if you want. 

After the release, I'll add eslint to prevent this kind of inconsistencies. Embrace yourselves for the incoming changes! KKK. :)